com.unity.netcode.gameobjects/Runtime/Spawning/NetworkPrefabHandler.cs

```diff
@@ -40,7 +40,8 @@ public interface INetworkPrefabInstanceHandler
         /// The most common approach is to make the <see cref="NetworkObject"/> inactive by calling <see cref="GameObject.SetActive(bool)"/>.
         /// </summary>
         /// <param name="networkObject">The <see cref="NetworkObject"/> being destroyed</param>
-        void Destroy(NetworkObject networkObject);
+        /// <returns>(true) destroy the parent <see cref="GameObject"/> (false) do not destroy the parent <see cref="GameObject"/> </returns>
```

Comment:
```suggestion
        /// <returns>(true) destroy the <see cref="GameObject"/> of this <see cref="NetworkObject"/> (false) do not destroy the<see cref="GameObject"/> </returns>
```
Same as above

Reply:
This I think we need to specify what we mean in a literal sense:
```
            var gobj = networkObject.gameObject;

            if (destroyGameObject && gobj != null)
            {
                if (NetworkManager.PrefabHandler.ContainsHandler(networkObject))
                {
                    // If the custom prefab handler returns true then we destroy the parent GameObject
                    // otherwise the custom prefab handler handles how they want to deal with despawning
                    // the NetworkObject
                    if(NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject))
                    {
                        if (SpawnedObjects.Remove(networkObject.NetworkObjectId))
                        {
                            SpawnedObjectsList.Remove(networkObject);
                        }
                        UnityEngine.Object.Destroy(gobj);
                        return;
```
We literally are destroying the GameObject that has the NetworkObject component attached to it.  You can destroy a component and the GameObject holding the component (i.e. parent  or perhaps owner GameObject?) will persist, but when you destroy the GameObject then all components and anything considered a "child" of the GameObject goes with it.

Does this make sesne?

Reply:
I don't think we should care about destroying the parent here, at all.
It could be inconvenient for users to think a little more about which component/gameobject they are destroying in the hierarchy but nevertheless, I don't think we should take on this responsibility here.
(we might lead by example somewhere else)

Reply:
I think the problem is how we use the word parent. When we have a `component` the corresponding `GameObject` is **NOT** a parent. Parent is a term we use for the hierarchy of GameObjects itself.

Reply:
This no longer returns a bool and that language no longer exists so I marked this as resolved.